### ISSUE TITLE
auc calculation issue

### ISSUE THREAD

**Issue by liangbright** [MedMNIST/MedMNIST#4]
Dear authors:

great work!

auc calculation is significantly different from sklearn document
https://scikit-learn.org/stable/modules/generated/sklearn.metrics.roc_auc_score.html#sklearn.metrics.roc_auc_score

for binary classification, why set a threshold 0.5?

**Comment by duducheng** [comment]
Hi @liangbright , 

Thanks for figuring it out!

Indeed we are implementing the standard AUC calculation. The reason why we did not use the standard functions in sklearn for multi-label / multi-class is the sklearn version. sklearn supports multi-class AUC only in recent sklearn versions. We add this function with easy code to support more users.

As for the binary class, we are sorry to introduce this bug in the public codebase. Yes, the AUC should not be with thresholding (just as that in multi-class and multi-label cases). We will fix this very soon.
